### PR TITLE
Moves may only have one aura boost

### PR DESCRIPTION
**Pull request by urkerab**
In case there are two Xerneas (or two Yveltal) on the field at once.

### CODE REVIEW

File: data/abilities.js [hunk 1]
```diff
@@ -579,9 +579,14 @@ exports.BattleAbilities = {
...
-		onAnyBasePower: function (basePower, source, target, move) {
+		onAnyTryPrimaryHit: function (target, source, move) {
 			if (target === source || move.category === 'Status' || move.type !== 'Dark') return;
-			return this.chainModify([move.hasAuraBreak ? 0x0C00 : 0x1547, 0x1000]);
+			move.auraBoost = 0x1547;
+		},
+		onAnyBasePower: function (basePower, source, target, move) {
+			if (!move.auraBoost) return;
+			this.chainModify([move.auraBoost, 0x1000]);
+			delete move.auraBoost;
```
Comment (Zarel) [inline data/abilities.js]: There's no reason to delete it, is there?

Comment (Zarel) [reply data/abilities.js]: You can just... set it, and then just return if it's truthy?

Comment (urkerab) [reply data/abilities.js]: Aura is a tricky ability, as it only gains one boost no matter how many Auras are on the field, and you don't have to have the Aura yourself. This means that you need some way of making sure only one Aura actually ends up applying the boost.

Comment (Zarel) [reply data/abilities.js]: > You can just... set it, and then just return if it's truthy?

That would be the way of making sure only one aura actually ends up applying the boost.

Comment (Slayer95) [reply data/abilities.js]: I don't understand your proposal either, @Zarel 

We already return if falsey...

Comment (Zarel) [reply data/abilities.js]: 2923b9414021615214 is my proposal.

Comment (Zarel) [reply data/abilities.js]: For the record, replacing `delete move.auraBoost` with `move.auraBoost = null` would at least be an improvement for V8 perf, but my approach also makes it possible for other effects to detect whether the move has been aura-boosted, and is fewer lines and is easier to read (at least to me).

Comment (urkerab) [reply data/abilities.js]: Should `CONTRIBUTING.md` mention constructs to avoid such as `delete`?

Comment (Zarel) [reply data/abilities.js]: Yeah.

**Pull request closed without merging**